README.md

```diff
@@ -2,31 +2,899 @@
+            System.out.println(publisher.getAddress());
+            System.out.println(publisher.getBooks());
+    
+            // get address data
+            Address address = entityManager.find(Address.class, 3);
+            System.out.println(address);
+            System.out.println(address.getPublisher());
+    
+            // get author data
+            Author author = entityManager.find(Author.class, 7);
+            System.out.println(author);
+            System.out.println(author.getBooks());
+    
+            // get book data
+            Book book = entityManager.find(Book.class, 9);
+            System.out.println(book);
+            System.out.println(book.getPublisher());
+            System.out.println(book.getAuthors());
+    
+            // close entity manager and factory
+            entityManager.close();
+            entityManagerFactory.close();
+        }
+    }
+    ```
+2. Update `persistence.xml` to set the `hibernate.hbm2ddl.auto` property to `none`.
+   Recall that this prevents Hibernate from deleting the existing table data.    
+   ```xml
+   <property name="hibernate.hbm2ddl.auto" value="none" /> <!-- create / create-drop / update / none -->
+   ```
+3. Execute the `JpaRead.main` method to query the database tables using JPA.
+4. The output from the print statements is interwoven with the Hibernate SQL commands.
+   We can suppress the SQL output by editing `persistence.xml` and setting the `hibernate.show_sql` property to `false`.    
+   ```xml
+   <property name="hibernate.show_sql" value="false" /> <!-- Show SQL in console -->
+   ```
+5. Now when we execute the `JpaRead.main` method to query the database tables using JPA,
+   the output is concise:
+
+```text
+Publisher{id=1, name='HarperCollins', website='harpercollins.com'}
+Address{id=3, street='null', city='New York', state='NY', zip='null', country='United States'}
+[Book{id=9, title='The Sour Grape', isbn='9780063045415', publicationDate=2022-11-01, pages=40, edition=0}, Book{id=12, title='Where the Sidewalk Ends', isbn='9780060256678', publicationDate=2014-02-18, pages=176, edition=1}, Book{id=10, title='The Giving Tree', isbn='9780060256654', publicationDate=2014-02-18, pages=64, edition=0}]
+Address{id=3, street='null', city='New York', state='NY', zip='null', country='United States'}
+Publisher{id=1, name='HarperCollins', website='harpercollins.com'}
+Author{id=7, firstName='Shel', lastName='Silverstein'}
+[Book{id=12, title='Where the Sidewalk Ends', isbn='9780060256678', publicationDate=2014-02-18, pages=176, edition=1}, Book{id=10, title='The Giving Tree', isbn='9780060256654', publicationDate=2014-02-18, pages=64, edition=0}]
+Book{id=9, title='The Sour Grape', isbn='9780063045415', publicationDate=2022-11-01, pages=40, edition=0}
+Publisher{id=1, name='HarperCollins', website='harpercollins.com'}
+[Author{id=6, firstName='Pete', lastName='Oswald'}, Author{id=5, firstName='Jory', lastName='John'}]
+```
 
 ## Conclusion
```

Comment:
I really like this conclusion. I might encourage students to print it out and tape it to their desks.

Reply:
lol